.github/workflows/usage-metrics-dashboard.yml

```diff
@@ -0,0 +1,93 @@
+name: Usage Metrics Dashboard
+
+# Re-usable workflows use the `workflow_call` trigger
+# https://docs.github.com/en/actions/sharing-automations/reusing-workflows#creating-a-reusable-workflow
+on:
+  workflow_call:
+
+# Setup the environment with the extension name for easy re-use
+# Also set the GH_TOKEN for the release-extension action to be able to use gh
+env:
+  EXTENSION_NAME: usage-metrics-dashboard
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  extension:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./extensions/${{ env.EXTENSION_NAME }}
+
+    steps:
+      # Checkout the repository so the rest of the actions can run with no issue
+      - uses: actions/checkout@v4
+
+      # We want to fail quickly if the linting fails, do that first
+      - uses: ./.github/actions/lint-extension
+        with:
+          extension-name: ${{ env.EXTENSION_NAME }}
+
+      # ---
+      # Run R tests
+      # ---
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: '4.3.3'
+
+      - uses: r-lib/actions/setup-renv@v2
+        with:
+          working-directory: extensions/${{ env.EXTENSION_NAME }}
+
+      - run: Rscript -e 'install.packages(c("testthat"))'
+        working-directory: extensions/${{ env.EXTENSION_NAME }}
+
+      - run: Rscript -e 'testthat::test_local()'
+        working-directory: extensions/${{ env.EXTENSION_NAME }}
+
+      # Now that the extension is built we need to upload an artifact to pass
+      # to the package-extension action that contains the files we want to be
+      # included in the extension
+      # This only includes necessary files for the extension to run leaving out
+      # the files that were used to build the /dist/ directory
+      - name: Upload built extension
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.EXTENSION_NAME }}
+          # Replace the below with the files your content needs
+          path: |
+            extensions/${{ env.EXTENSION_NAME }}/app.R
+            extensions/${{ env.EXTENSION_NAME }}/R/get_usage.R
+            extensions/${{ env.EXTENSION_NAME }}/R/integrations.R
+            extensions/${{ env.EXTENSION_NAME }}/R/ui_components.R
+            extensions/${{ env.EXTENSION_NAME }}/R/visit_processing.R
```

Comment:
You could include the whole directory with a single line if that would be easier?

Reply:
Done. Thanks!